src/box/space.c

```diff
@@ -1430,6 +1431,21 @@ space_execute_dml(struct space *space, struct txn *txn,
 		if (space->vtab->execute_upsert(space, txn, request) != 0)
 			return -1;
 		break;
+	case IPROTO_INSERT_ARROW:
+		*result = NULL;
+		struct ArrowArray array;
+		struct ArrowSchema schema;
+		if (arrow_ipc_decode(&array, &schema, request->arrow,
+				     request->arrow_end) != 0)
+			return -1;
+		assert(array.release != NULL);
+		assert(schema.release != NULL);
+		if (space->vtab->execute_insert_arrow(space, txn, &array,
+						      &schema) != 0)
+			return -1;
+		array.release(&array);
+		schema.release(&schema);
```

Comment:
Should one release the `array` and `schema` prior to the `return -1`?

Reply:
Yes, thanks.

Reply:
Nice approach! I'd do it too verbose.